IntegratedBreedingPlatform/APITests
# Fix brapi services

**List of Changes:**

1. add test to verify max pageSize on location-resource-brapi
2. add tests for GET ​/{crop}​/brapi​/v1​/variables
3. add nursery1StudyDbId variable (holds studyDbId of the study - Study with Imported Cross Nursery 2018002
4. add tests for GET ​/{crop}​/brapi​/v1​/studies​/{studyDbId}​/observationunits
5. add tests for GET /{crop}​/brapi​/v1​/studies​/{studyDbId}​/observationvariables
6. organize tests of program-resource-brapi to a subfolder
7. remove deprecated tests that passes diff formats (format parameter field no longer exist in the call GET ​/{crop}​/brapi​/v1​/programs)
8. add test that verify pagination and progamName parameter
9. rename tests for GET ​/{crop}​/brapi​/v1​/studies​/{studyDbId}​/table
10. organize study-brapi-resource tests into subfolders
11. add tests for GET /{crop}/brapi/v2/studies/{studyDbId}
13. Use env variable for passing non existing studyId to GET​/{crop}​/brapi​/v1​/trials​/{trialDbId}​/table
14. add subfolder for brapi-trial tests and organize existing tests
15. add test to verify error when entered pageSize greater than maximum for GET ​/{crop}​/brapi​/v1​/trials​/{trialDbId}​/table


Tasks | /
-- | --
**Pre-merging** |  
Followed basic standards | ✅ 
Ensured that the List of API Services and Calls spreadsheet is updated |  ✅ 
Ensured that all test data created and used in the tests are added in the Setting up Initial Data (Data Seeding) page |  ✅ 
Ensured that the entire test suite is passing in branch using PostmanCollections-CIRun or PostmanCollections-PerfTests (if changes are related to performance test) jenkins jobs |  ✅ 
Ensured that Testlink is updated |  ✅ 
**Post-merging** |  
Ensured that the entire test suite is passing in master using PostmanCollections-CIRun and PostmanCollections-PerfTests (if changes are related to performance test) jenkins jobs |  
Deleted branch |
